extensions/community_extensions/dsa_transparency.md

```diff
@@ -0,0 +1,379 @@
+
+<h2>Introduction</h2>
+<p>The Digital Services Act (DSA) was adopted in October 2022, and the date of applicability for Platform companies is 16 February 2024.&nbsp; Along with the Digital Markets Act (DMA), the DSA is intended to improve the confidence of both private consumers and business users of Online Platforms in the products and services they access via those platforms, as well as the advertising they are exposed to on them, and to ensure a level playing field between platforms.&nbsp; The DSA lays down transparency obligations in relation to advertising; these obligations apply to Online Platforms, &ldquo;Very Large Online Platforms&rdquo; (VLOPs), and &ldquo;Very Large Online Search Engines&rdquo; (VLOSEs) as defined by the Digital Services Act.&nbsp;&nbsp;</p>
+<p>Article 26 of the DSA requires Online Platforms to ensure that users have real-time access to certain elements of information about any ad shown to them on an Online Platform:</p>
+<ul>
+<li>That the ad is indeed an ad;</li>
+<li>The identity of the advertiser;</li>
+<li>The identity of the party that financed the ad, if it is different from the advertiser;</li>
+<li>Information about the &ldquo;main parameters&rdquo; used to select the ad presented to the end-user;</li>
+<li>Where applicable, information about any means users may have at their disposal to change those main parameters.&nbsp;&nbsp;</li>
+</ul>
+<p>Although the legal obligation to provide the user-facing information disclosures applies to Online Platforms, it is clear that in many advertising scenarios, those platforms will need to rely on third-party vendors for the data that will be required to populate the disclosures.&nbsp; To ensure that the third parties are equipped to provide this support, this technical specification standardizes the collection, compilation and transport of the data, leaving Online Platforms free to decide how they wish to make the user-facing disclosures, including if they want to delegate the disclosures to another party.&nbsp;&nbsp;</p>
+<p>This technical specification provides data formats and transport for the advertising industry to implement relevant DSA transparency information. This solution should be applicable across most relevant use cases including; programmatic and non-programmatic media buys, channels including desktop web, mobile (web/app), video, CTV.</p>
+<h2>Object Specification for OpenRTB 2.X</h2>
+<h3>Bid Request</h3>
+<h4>Object: Regs</h4>
+<div>
+<table>
+<tbody>
+<tr>
+<td><strong>Attribute</strong></td>
+<td><strong>Type</strong></td>
+<td><strong>Description</strong></td>
+</tr>
+<tr>
+<td><code>ext.dsa</code></td>
+<td>object&nbsp;</td>
+<td>Extension for DSA transparency information</td>
+</tr>
+</tbody>
+</table>
+</div>
+<h4>Object: DSA</h4>
+<div>
+<table>
+<tbody>
+<tr>
+<td><strong>Attribute</strong></td>
+<td><strong>Type</strong></td>
+<td><strong>Description</strong></td>
+</tr>
+<tr>
+<td><code>required</code></td>
```

Comment:
[critical]

Word `required` is a reserved one in many programming languages as well as in protobuf v2 and thrift:
* proto v2 https://protobuf.dev/reference/protobuf/proto2-spec/#fields
* thrift https://thrift.apache.org/docs/idl.html#field-requiredness
Please consider to rename it to something else, otherwise it will be very painful to implement json parsing.

```suggestion
<td><code>requireness</code></td>
```

Reply:
How about req

Reply:
Or `mandatory`

Reply:
+1 please do not use `required`.